src/main/java/org/spongepowered/server/launch/plugin/asm/PluginAnnotationVisitor.java

```diff
@@ -57,34 +55,54 @@ String getAnnotation() {
         return "@Plugin";
     }
 
-    private void checkState(State state) {
-        Preconditions.checkState(this.state == state, "Expected state %s, but is %s", state, this.state);
-    }
-
     @Override
     public void visit(String name, Object value) {
         if (this.state == State.AUTHORS) {
+            if (!(value instanceof String)) {
+                throw new InvalidPluginException("Plugin annotation has invalid element 'author'");
+            }
             this.metadata.addAuthor((String) value);
             return;
         }
 
-        checkState(State.DEFAULT);
-        checkNotNull(name, "name");
+        if (name == null) {
+            throw new InvalidPluginException("Plugin annotation has null element");
```

Comment:
Can you clarify that `name` is null, e.g. `Plugin annotation attribute name is null`